apps/distgo/params/docker.go

```diff
@@ -15,37 +15,44 @@
 package params
 
 import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+
 	"github.com/pkg/errors"
 )
 
 type DockerDepType string
+type DockerImageType string
 
 const (
-	DockerDepSLS    DockerDepType = "sls"
-	DockerDepBin    DockerDepType = "bin"
-	DockerDepRPM    DockerDepType = "rpm"
-	DockerDepDocker DockerDepType = "docker"
+	DockerDepSLS          DockerDepType = "sls"
+	DockerDepBin          DockerDepType = "bin"
+	DockerDepRPM          DockerDepType = "rpm"
+	DockerDepDocker       DockerDepType = "docker"
+	ManifestLabel         string        = "com.palantir.sls.manifest"
+	ConfigurationLabel    string        = "com.palantir.sls.configuration"
+	ConfigurationFileName string        = "configuration.yml"
 )
 
 type DockerDep struct {
 	Product    string
 	Type       DockerDepType
 	TargetFile string
 }
-type DockerDeps []DockerDep
 
-func (d DockerDeps) ToMap() map[string]map[DockerDepType]string {
-	m := make(map[string]map[DockerDepType]string)
-	for _, dep := range d {
-		if m[dep.Product] == nil {
-			m[dep.Product] = make(map[DockerDepType]string)
-		}
-		m[dep.Product][dep.Type] = dep.TargetFile
-	}
-	return m
+type DockerImage interface {
+	SetRepository(repo string)
+	SetDefaults(repo string, tag string)
+	Coordinates() (string, string)
+	ContextDirectory() string
+	Dependencies() []DockerDep
+	Build(buildSpec ProductBuildSpecWithDeps) error
 }
 
-type DockerImage struct {
+type SimpleDockerImage struct {
```

Comment:
Not sure if `Simple` is the best naming here -- maybe `DefaultDockerImage` or `StandardDockerImage` instead? Not totally set on this, though (may be okay to leave as-is)